### PR TITLE
JavaScript refactor: separate out article bootstrap from app.js

### PR DESCRIPTION
**Pull request by robertberry-zz**
Obviously I need to test the heck out of this

It splits some of the trail-only stuff out of `common.js` and puts it into `trail.js`, which is then called by the other trail bootstraps. This means a lot of trail-only code won't be included in the fronts JavaScript.

### CODE REVIEW

**Comment by piuccio** [comment]
I would have created a `robust.all` method inside `robust` instead of adding a new `robusts` utility, but it's a matter of style.

It's sometime confusing that some modules look like this

```
robust(...);
robust(...);
robust(...);
trail();
```

Initially I though you forgot to wrap trail inside robust, so I had to check inside trail to realise that it's using robust internally. Maybe you can add a comment in this case.

It's also interesting that you modified some file to return a function instead of an object with `init`. It's getting hard to understand when a file requires an init and when it doesn't. There's little consistency in the project.

Otherwise looks good. I'll let someone from QA to +1

**Comment by robertberry** [comment]
@piuccio I really don't understand why there are so many instances of people wrapping functions in objects when it's not necessary. I assume it's because historically people haven't understood first class functions properly.

File: common/app/views/fragments/javaScriptLaterSteps.scala.html [hunk 1]
```diff
@@ -230,7 +230,11 @@
                 'bootstraps/image-content': '@Static("javascripts/bootstraps/image-content.js")',
                 'bootstraps/membership':    '@Static("javascripts/bootstraps/membership.js")',
                 'bootstraps/sudoku':        '@Static("javascripts/bootstraps/sudoku.js")',
-                'bootstraps/video-player':  '@Static("javascripts/bootstraps/video-player.js")'
+                'bootstraps/video-player':  '@Static("javascripts/bootstraps/video-player.js")',
+                'bootstraps/article': '@Static("javascripts/bootstraps/article.js")',
+                'bootstraps/liveblog': '@Static("javascripts/bootstraps/liveblog.js")',
+                'bootstraps/trail': '@Static("javascripts/bootstraps/trail.js")',
+                'bootstraps/gallery': '@Static("javascripts/bootstraps/gallery.js")'
```
Comment (OliverJAsh) [inline common/app/views/fragments/javaScriptLaterSteps.scala.html]: Do you need to create jspm bundles for these too? See `bundle.js`.

RequireJS will be gone soon so we won't have to keep supporting both module loaders.

File: static/src/javascripts/bootstraps/app.js [hunk 2]
```diff
@@ -20,20 +16,16 @@ define([
...
     sport,
     SaveForLaterTest
 ) {
 
-    var bootstrapContext = function (featureName, boostrap) {
+    var bootstrapContext = function (featureName, bootstrap) {
             raven.context(
                 { tags: { feature: featureName } },
-                boostrap.init,
+                bootstrap.init,
```
Comment (OliverJAsh) [inline static/src/javascripts/bootstraps/app.js]: I preferred `boostrap` :smile_cat:

**Comment by OliverJAsh** [comment]
Brilliant work. :+1: Merge conflict BTW.

Please do test with jspm :smile:

**Comment by OliverJAsh** [comment]
I think this happened because we have very little visibility of our dependency graph. Perhaps there's a tool that will visually print the dependency graph.

**Comment by piuccio** [comment]
There is a tool that visually prints the dependency graph, it's a huge
mess. Good luck trying to figure out something out of it

On Tue, 23 Jun 2015 at 15:35 Oliver Joseph Ash <notifications@github.com>
wrote:

> I think this happened because we have very little visibility of our
> dependency graph. Perhaps there's a tool that will visually print the
> dependency graph.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/guardian/frontend/pull/9574#issuecomment-114527704>.
>

-- 

------------------------------
Visit theguardian.com. On your mobile and tablet, download the Guardian 
iPhone and Android apps theguardian.com/guardianapp and our tablet editions 
theguardian.com/editions.  Save up to 57% by subscribing to the Guardian 
and Observer - choose the papers you want and get full digital access. 
 Visit subscribe.theguardian.com

This e-mail and all attachments are confidential and may also be 
privileged. If you are not the named recipient, please notify the sender 
and delete the e-mail and all attachments immediately. Do not disclose the 
contents to another person. You may not use the information for any 
purpose, or store, or copy, it in any way.  Guardian News & Media Limited 
is not liable for any computer viruses or other material transmitted with 
or as part of this e-mail. You should employ virus checking software.
 
Guardian News & Media Limited is a member of Guardian Media Group plc. Registered 
Office: PO Box 68164, Kings Place, 90 York Way, London, N1P 2AP.  Registered 
in England Number 908396

File: static/src/javascripts/bootstraps/identity-common.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
+/**
+ * Identity crap that has to run on every page (putting in usernames, avatars, etc.)
+ *
+ * Explicitly NOT stuff that only runs on the identity pages. Put that in profile.js or I will hunt you down. I WILL
+ * HUNT YOU DOWN.
```
Comment (sndrs) [inline static/src/javascripts/bootstraps/identity-common.js]: :fearful:

Comment (gklopper) [inline static/src/javascripts/bootstraps/identity-common.js]: :+1: reminds me of this toptier comment https://github.com/guardian/frontend/blob/master/common/app/model/meta.scala#L196-L210

**Comment by sndrs** [comment]
:+1: can't find any problems in identity so far

**Comment by OliverJAsh** [comment]
This works with jspm? No reason it wouldn't…

**Comment by OliverJAsh** [comment]
Nobody understands how these files are loaded, which is what lead to this situation.

I'll do some thinking about how we can restructure our files to increase visibility over how they are loaded.

**Comment by ScottPainterGNM** [comment]
@jennysivapalan This was the PR I mentioned

**Comment by jennysivapalan** [comment]
@ScottPainterGNM Great thanks, will take a look through

**Pull request merged**